### PR TITLE
Add NavbarRoute component for repetitive route NavbarButtons

### LINKED ISSUE #251: Clean up `NavbarButton` click handlers
Maybe `NavbarButton` should have a `href` property, which would clean up all these `on:click` handlers.

_Originally posted by @brandonbloom in https://github.com/deref/exo/pull/241#discussion_r695928130_

### PR DESCRIPTION
**Pull request by jwmza**
Fix #251

### CODE REVIEW

File: gui/src/components/nav/NavbarRoute.svelte [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<script lang="ts">
+  import NavbarButton from './NavbarButton.svelte';
+  import * as router from 'svelte-spa-router';
+
+  export let title: string | undefined = undefined;
```
Comment (brandonbloom) [inline gui/src/components/nav/NavbarRoute.svelte:5]: nit: `= undefined` can just be omitted:

```typescript
  export let title: string | undefined;
```

**Review by brandonbloom — APPROVED** [review]
LGTM, though I wonder why not enhance NavbarButton rather than wrapping it?

File: gui/src/components/nav/NavbarRoute.svelte (hunk 1 above)
Comment (jwmza) [reply gui/src/components/nav/NavbarRoute.svelte]: Good to know, thanks :)

**Comment by jwmza** [comment]
> I wonder why not enhance NavbarButton rather than wrapping it?

Could do it either way but I figured this was more elegant to allow a combination of link and custom click behaviors. Happy to change it to the other approach if there's a reason to.

**Pull request merged**